src/utils/ScreenReader.ts

```diff
@@ -0,0 +1,281 @@
+export interface SpeakOptions {
+  now?: boolean
+}
+
+export interface TextToSpeech {
+  /**
+   * Directly triggers speech of text. Resolves when speaking is done.
+   */
+  speak(text: string, options?: SpeakOptions): Promise<void>
+
+  /**
+   * Stops any speaking that is currently happening.
+   */
+  stop(): void
+}
+
+/**
+ * Implement this to provide screen reading.
+ */
+export interface ScreenReader {
+  /**
+   * Call this with an event target when a focus event occurs. Resolves when speaking is done.
+   */
+  onFocus(target?: EventTarget): Promise<void>
+
+  /**
+   * Call this with an event target when a click event occurs. Resolves when speaking is done.
+   */
+  onClick(target?: EventTarget): Promise<void>
+
+  /**
+   * Call this when a page load occurs. Resolves when speaking is done.
+   */
+  onPageLoad(): Promise<void>
+
+  /**
+   * Call this when the screen reader is enabled. Resolves when speaking is done.
+   */
+  onScreenReaderEnabled(): Promise<void>
+
+  /**
+   * Call this when the screen reader is disabled. Resolves when speaking is done.
```

Comment:
"Resolves when speaking is done." or immediately?

Reply:
Oh, I see. After speaking "Screen reader disabled" correct?

Reply:
> Oh, I see. After speaking "Screen reader disabled" correct?

Yep.